tests/image_tiff/util.rs

```diff
@@ -1,15 +1,78 @@
-use std::env::current_dir;
+use async_tiff::TIFF;
 use std::sync::Arc;
 
+#[cfg(feature = "object_store")]
 use async_tiff::reader::ObjectReader;
-use async_tiff::TIFF;
+#[cfg(feature = "object_store")]
 use object_store::local::LocalFileSystem;
+#[cfg(feature = "object_store")]
+use std::env::current_dir;
+
+#[cfg(not(any(feature = "tokio", feature = "object_store")))]
+use async_tiff::{
+    error::{AsyncTiffError, AsyncTiffResult},
+    reader::AsyncFileReader,
+};
+#[cfg(not(any(feature = "tokio", feature = "object_store")))]
+use bytes::Bytes;
+#[cfg(not(any(feature = "tokio", feature = "object_store")))]
+use futures::{future::BoxFuture, FutureExt};
+#[cfg(not(any(feature = "tokio", feature = "object_store")))]
+use std::ops::Range;
 
 const TEST_IMAGE_DIR: &str = "tests/image_tiff/images/";
 
+#[cfg(feature = "object_store")]
 pub(crate) async fn open_tiff(filename: &str) -> TIFF {
     let store = Arc::new(LocalFileSystem::new_with_prefix(current_dir().unwrap()).unwrap());
     let path = format!("{TEST_IMAGE_DIR}/{filename}");
     let reader = Arc::new(ObjectReader::new(store.clone(), path.as_str().into()));
     TIFF::try_open(reader).await.unwrap()
 }
+
+#[cfg(all(feature = "tokio", not(feature = "object_store")))]
+pub(crate) async fn open_tiff(filename: &str) -> TIFF {
+    // let store = Arc::new(LocalFileSystem::new_with_prefix(current_dir().unwrap()).unwrap());
+    let path = format!("{TEST_IMAGE_DIR}/{filename}");
+    let reader = Arc::new(
+        tokio::fs::File::open(path)
+            .await
+            .expect("could not open file"),
+    );
+    TIFF::try_open(reader).await.unwrap()
+}
```

Comment:
We shouldn't add this test, because all this is doing is testing that the same file can be read with tokio. But as long as the traits compile we can use only `object_store` at runtime.

We should always use `object_store` for testing, and so `object_store` should always be included in `dev-dependencies`.